docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,1069 @@
+│   ├── sampleB
+│   │   └── 1
+│   │       └── sampleB_stats.txt
+│   └── sampleC
+│       └── 1
+│           └── sampleC_stats.txt
+└── tumor
+    ├── sampleA
+    │   ├── 1
+    │   │   └── sampleA_stats.txt
+    │   └── 2
+    │       └── sampleA_stats.txt
+    ├── sampleB
+    │   └── 1
+    │       └── sampleB_stats.txt
+    └── sampleC
+        └── 1
+            └── sampleC_stats.txt
+```
+
+See how using sample metadata as values gives us powerful flexibility in our pipeline. By propagating metadata alongside our data in tuples, we can:
+
+1. Create organized output directories based on sample attributes
+2. Make decisions in processes based on sample properties
+3. Split, join, and recombine data based on metadata values
+
+This pattern of keeping metadata explicit and attached to the data (rather than encoded in filenames) is a core best practice in Nextflow that enables building robust, maintainable bioinformatics workflows.
+
+### Takeaway
+
+- Processes can take files as input and produce new files as output
+- The `publishDir` directive organizes outputs based on metadata values
+- Metadata in tuples enables structured organization of results
+- This approach creates maintainable workflows with clear data provenance
+
+---
+
+## 6. Using remote files
+
+One of the key features of Nextflow is the ability to transparently switch between local files (on the same machine) to remote files accessible over the internet. To do this, all you need to do as a user is switch the file path from a normal file path (e.g. `/path/to/data`) to a file path with a remote protocol at the start. For example, replacing `/path/to/data` with `s3://path/to/data` will switch to using the S3 protocol. Many different protocols are supported:
+
+- HTTP(S)/FTP (http://, https://, ftp://)
+- Amazon S3 (s3://)
+- Azure Blob Storage (az://)
+- Google Cloud Storage (gs://)
+
+To use these, simply replace the string and Nextflow will handle authentication and staging the files to the right place, downloading or uploading and all other file operations you would expect. We call this string the Uniform Resource Identifier (URI).
+
+The key strength of this is we can switch between environments without changing any pipeline logic. For example, you can develop with a small, local test set before moving to a remote set of data by changing the URI.
+
+### 6.1 Using a file from the internet
+
+!!! warning
+Accessing remote data requires an internet connection!
+
+In your workflow, replace the string path with an HTTPS one to download this file from the internet. We are going to swap the relative path of the FASTQ files with the remote one stored on the internet. This is the same data as we have been previously using.
+
+=== "After"
+
+    ```groovy title="main.nf" linenums="27"
+    ch_fastq = Channel.fromFilePairs('https://github.com/nextflow-io/training/blob/bb187e3bfdf4eec2c53b3b08d2b60fdd7003b763/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz')
```

Comment:
This would need to be a glob path, no?

Reply:
No, actually, we'd need to not use `fromFilePairs`, right?

Reply:
Yeah, but it only produces one file:

```
[sampleA_rep1_normal_R1_001, [/nextflow-io/training/blob/bb187e3bfdf4eec2c53b3b08d2b60fdd7003b763/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz]]
```

Somewhat defeats the point of `fromFilePairs`

Reply:
Yep, exactly my thinking when I suggested putting remove files earlier rather than as a bolt-on.